examples/customresources/demos/remoteserver/server_test.go

```diff
@@ -20,74 +21,115 @@ import (
-	pmgr := pexec.NewProcessManager(logger.Sublogger("process.inter").AsZap())
-	pCfg := pexec.ProcessConfig{
-		ID:      "Intermediate",
-		Name:    "go",
-		Args:    []string{"run", utils.ResolveFile("./web/cmd/server/main.go"), "-config", tmpConf.Name()},
-		CWD:     "",
-		OneShot: false,
-		Log:     true,
+		cfgServer, err := config.Read(ctx, utils.ResolveFile("./examples/customresources/demos/remoteserver/remote.json"), logger)
+		test.That(t, err, test.ShouldBeNil)
+		remoteB, err := robotimpl.New(ctx, cfgServer, logger.Sublogger("remoteB"))
+		test.That(t, err, test.ShouldBeNil)
+		options := weboptions.New()
+		options.Network.BindAddress = remoteAddrB
+
+		err = remoteB.StartWeb(ctx, options)
+		if err != nil && strings.Contains(err.Error(), "address already in use") {
+			logger.Infow("Port in use. Restarting on new port.", "port", port, "err", err)
+			test.That(t, remoteB.Close(context.Background()), test.ShouldBeNil)
+			continue
+		} else {
+			test.That(t, err, test.ShouldBeNil)
+			defer func() {
+				test.That(t, remoteB.Close(context.Background()), test.ShouldBeNil)
+			}()
+			break
+		}
 	}
-	_, err = pmgr.AddProcessFromConfig(context.Background(), pCfg)
-	test.That(t, err, test.ShouldBeNil)
-	err = pmgr.Start(context.Background())
-	defer func() {
-		test.That(t, pmgr.Stop(), test.ShouldBeNil)
-	}()
-	test.That(t, err, test.ShouldBeNil)
 
-	remoteConfig := &config.Config{
+	// Create remote A. Loop to ensure we find an available port.
+	var remoteAddrA string
+	success := false
+	for portTryNum := 0; portTryNum < 10; portTryNum++ {
+		// The process executing this test has loaded the "gizmo" API + model into a global registry
+		// object. We want this intermediate remote to be unaware of the custom gizmo resource. We
+		// start up a separate viam-server process to achieve this.
+		port, err := goutils.TryReserveRandomPort()
+		test.That(t, err, test.ShouldBeNil)
+		remoteAddrA = fmt.Sprintf("localhost:%d", port)
+
+		tmpConf, err := os.CreateTemp(t.TempDir(), "*.json")
+		test.That(t, err, test.ShouldBeNil)
+		_, err = tmpConf.WriteString(fmt.Sprintf(
+			`{"network":{"bind_address":"%s"},"remotes":[{"address":"%s","name":"robot1"}]}`,
+			remoteAddrA, remoteAddrB))
+		test.That(t, err, test.ShouldBeNil)
+		err = tmpConf.Sync()
+		test.That(t, err, test.ShouldBeNil)
+
+		processLogger, logObserver := logging.NewObservedTestLogger(t)
+		pmgr := pexec.NewProcessManager(processLogger.Sublogger("remoteA").AsZap())
+		pCfg := pexec.ProcessConfig{
+			ID:      "Intermediate",
+			Name:    "go",
+			Args:    []string{"run", utils.ResolveFile("./web/cmd/server/main.go"), "-config", tmpConf.Name()},
+			CWD:     "",
+			OneShot: false,
+			Log:     true,
+		}
+		_, err = pmgr.AddProcessFromConfig(context.Background(), pCfg)
+		test.That(t, err, test.ShouldBeNil)
+		err = pmgr.Start(context.Background())
+		test.That(t, err, test.ShouldBeNil)
+		if success = robottestutils.WaitForServing(logObserver, port); success {
```

Comment:
After writing the loop a couple times, I got in my groove and "simplified" the success stuff by a line or two. I hope the minor inconsistency isn't problematic.